pytensor/tensor/rewriting/subtensor.py

```diff
@@ -381,6 +382,57 @@ def local_useless_slice(fgraph, node):
             return [node.inputs[0]]
 
 
+@register_useless
+@register_canonicalize
+@register_stabilize
+@register_specialize
+@node_rewriter([Subtensor])
+def local_replace_slice(fgraph, node):
+    """
+    Rewrite Subtensor of the form:
+        1. X[0:7:1] -> X[None:None:None]
+        2. X[0:-1:2] -> X[None:6:2]
+
+    """
+    idxs = get_idx_list(node.inputs, node.op.idx_list)
+    x = node.inputs[0]
+
+    if not idxs:
+        return [x]
+
+    new_idxs = list(idxs)
+    idx_flag = False
+    for dim, s in enumerate(new_idxs):
+        if not isinstance(s, slice):
+            continue
+
+        idx_change = False
+        start = s.start
+        stop = s.stop
+        step = s.step
+        if extract_constant(start, only_process_constants=True) == 0:
+            idx_change = True
+            start = None
+
+        if extract_constant(stop, only_process_constants=True) == x.type.shape[dim]:
+            idx_change = True
+            stop = None
+
+        if extract_constant(step, only_process_constants=True) == 1:
+            idx_change = True
+            step = None
+
+        if idx_change:
+            idx_flag = True
+            new_idxs[dim] = slice(start, stop, step)
+
+    if idx_flag is True:
+        return [x[tuple(new_idxs)]]
+    else:
+        # Subtensor is not needed at all
+        return [x]
```

Comment:
This seems wrong, it just meant there was nothing you could change about the indexes, not that they weren't needed